src/engine/serverbrowser.cpp

```diff
@@ -218,35 +220,13 @@ int connectwithtimeout(ENetSocket sock, const char *hostname, const ENetAddress
 }
 
 vector<serverinfo *> servers;
-bool sortedservers = true;
 ENetSocket pingsock = ENET_SOCKET_NULL;
 int lastinfo = 0;
 
-static serverinfo *newserver(const char *name, int port = SERVER_PORT, int priority = 0, const char *desc = NULL, const char *handle = NULL, const char *flags = NULL, const char *branch = NULL, uint ip = ENET_HOST_ANY)
-{
-    serverinfo *si = new serverinfo(ip, port, priority);
-
-    if(name) copystring(si->name, name);
-    else if(ip == ENET_HOST_ANY || enet_address_get_host_ip(&si->address, si->name, sizeof(si->name)) < 0)
-    {
-        delete si;
-        return NULL;
-    }
-    if(desc && *desc) copystring(si->sdesc, desc, MAXSDESCLEN+1);
-    if(handle && *handle) copystring(si->authhandle, handle);
-    if(flags && *flags) copystring(si->flags, flags);
-    if(branch && *branch) copystring(si->branch, branch, MAXBRANCHLEN+1);
-
-    servers.add(si);
-    sortedservers = false;
-
-    return si;
-}
-
 void addserver(const char *name, int port, int priority, const char *desc, const char *handle, const char *flags, const char *branch)
 {
-    loopv(servers) if(!strcmp(servers[i]->name, name) && servers[i]->port == port) return;
-    if(newserver(name, port, priority, desc, handle, flags, branch) && verbose >= 2)
+    loopv(servers) if( servers[i]->is_same( name, port ) ) return;
```

Comment:
I'm not a big fan of this kind of method. Maybe it's also just the naming that isn't entirely clear.